owl/sys.scm

```diff
@@ -259,26 +262,66 @@
       (sc O_SYNC 108)
       (sc O_TRUNC 109)
       (sc O_TTY_INIT 110)
+      (sc O_ACCMODE 111)
+      (sc FD_CLOEXEC 112)
+      (sc F_DUPFD 113)
+      (sc F_DUPFD_CLOEXEC 114)
+      (sc F_GETFD 115)
+      (sc F_SETFD 116)
+      (sc F_GETFL 117)
+      (sc F_SETFL 118)
+      (sc F_GETOWN 119)
+      (sc F_SETOWN 120)
+      (sc F_GETLK 121)
+      (sc F_SETLK 122)
+      (sc F_SETLKW 123)
+      (sc F_RDLCK 124)
+      (sc F_UNLCK 125)
+      (sc F_WRLCK 126)
 
       (define (close fd)
          (sys 2 fd))
 
+      (define (fcntl port cmd arg)
+         (sys 15 port cmd arg))
+
+      (define (toggle-file-status-flag port flag on?)
+         (let ((flags (fcntl port (F_GETFL) 0)))
```

Comment:
`(let ((foo binding)) (and foo body))` can also be handled as `(if-lets ((foo binding)) body)`. The body is evaluated iff all first values of the bindings are not false.

This could otherwise be called `if-let*`, but Scheme `let*` is not defined for multiple values.